src/core/models/aggregate.cc

```diff
@@ -35,60 +35,106 @@
+
+For one- and two-column frames the aggregation is based on
+the standard equal-interval binning for numeric columns, and
+grouping for string columns.
+
+When the input frame has more columns than two,
+a parallel one-pass Ad-Hoc algorithm is employed, see description of
+`Aggregator<T>::group_nd() <https://github.com/h2oai/datatable/blob/master/src/core/models/aggregator.cc>`_
+method for more details. This algorithm takes into account
+the numeric columns only, and all the string columns are ignored.
+
 
 Parameters
 ----------
 frame: Frame
-    Frame to be aggregated.
+    The input frame containing numeric or string columns.
+
 min_rows: int
-    Minimum number of rows a datatable should have to be aggregated.
-    If datatable has `nrows` that is less than `min_rows`, aggregation
-    is bypassed, and all rows become exemplars.
+    Minimum number of rows the input frame should have to be aggregated.
+    If `frame` has less rows than `min_rows`, aggregation
+    is bypassed, in the sence that all the input rows become exemplars.
+
 n_bins: int
     Number of bins for 1D aggregation.
+
 nx_bins: int
-    Number of x bins for 2D aggregation.
+    Number of bins for the first column for 2D aggregation.
+
 ny_bins: int
-    Number of y bins for 2D aggregation.
+    Number of bins for the second column for 2D aggregation.
+
 nd_max_bins: int
-    Maximum number of exemplars for ND aggregation, not a hard limit.
+    Maximum number of exemplars for ND aggregation. It is guaranteed
+    that the ND algorithm will return less than `nd_max_bins` exemplars,
+    but the exact number may vary from run to run due to parallelization.
+
 max_dimensions: int
-    Number of columns at which start using the projection method.
+    Number of columns at which the projection method is used for
+    ND aggregation.
+
 seed: int
     Seed to be used for the projection method.
+
 double_precision: bool
-    Whether to use double precision arithmetic or not.
+    An option to indicate whether double precision, i.e. `float64`,
+    or single precision, i.e. `float32`, arithmetic should be used
+    for computations.
+
 fixed_radius: float
     Fixed radius for ND aggregation, use it with caution.
     If set, `nd_max_bins` will have no effect and in the worst
     case number of exemplars may be equal to the number of rows
     in the data. For big data this may result in extremly large
-    execution times. Since all the columns are normalized to [0, 1),
+    execution times. Since all the columns are normalized to `[0, 1)`,
     the `fixed_radius` value should be choosen accordingly.
 
-Returns
--------
-A list `[frame_exemplars, frame_members]`, where
-- `frame_exemplars` is the aggregated `frame` with an additional
-  `members_count` column, that specifies number of members for each exemplar.
-- `frame_members` is a one-column datatable that contains `exemplar_id` for
-  each row from the original `frame`.
-)"
+return: List[Frame, Frame]
```

Comment:
This is not a valid type. You can have either `Tuple[Frame, Frame]` or `List[Frame]`. I guess in this case it would make more sense to change the return type to be a tuple.

Reply:
Why it is not valid?